### PR TITLE
Check makefile dependencies

### PR DESCRIPTION
**Pull request by itaiad200**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,3 +1,7 @@
+# check mandatory dependencies
+EXECUTABLES = go docker npm
```
[the remark this answers is not in the document]
Comment (nopcoder) [reply Makefile:2]: validate static names will prevent other users override the values in the execution time

[the remark this answers is not in the document]
Comment (itaiad200) [reply Makefile]: Pretty sure Oz meant the `statik` pkg. Added.

File: Makefile [hunk 2]
```diff
@@ -1,11 +1,18 @@
-
 GOCMD=$(shell which go)
 DOCKER=$(shell which docker)
 GOBINPATH=$(shell $(GOCMD) env GOPATH)
 NPM=$(shell which npm)
 STATIK=$(GOBINPATH)/bin/statik
 GOLANGCILINT_VERSION=v1.25.1
 
+# check for mandatory dependencies - binaries can be overridden.
+$(if $(GOCMD),,$(error "Missing dependency - no go in PATH"))
```
Comment (nopcoder) [inline Makefile:9]: s/go/Go/

File: Makefile [hunk 3]
```diff
@@ -1,11 +1,18 @@
...
+$(if $(DOCKER),,$(error "Missing dependency - no docker in PATH"))
```
Comment (nopcoder) [inline Makefile:10]: s/docker/Docker/

File: Makefile [hunk 4]
```diff
@@ -1,11 +1,18 @@
-
 GOCMD=$(shell which go)
 DOCKER=$(shell which docker)
 GOBINPATH=$(shell $(GOCMD) env GOPATH)
 NPM=$(shell which npm)
 STATIK=$(GOBINPATH)/bin/statik
 GOLANGCILINT_VERSION=v1.25.1
 
+# check for mandatory dependencies - binaries can be overridden.
```
Comment (nopcoder) [inline Makefile:8]: general question - why do we perform this check always. If I use the makefile to generate docs I am not required to have all these tools.
can you condition the check to the stop or add another step to check requirements?

**Pull request merged**